osu.Game.Tournament/Components/TournamentBeatmapPanel.cs

```diff
@@ -192,5 +185,86 @@ private void updateState()
                 Alpha = 1;
             }
         }
+
+        private class ModSprite : Container
+        {
+            public string Mod;
+
+            public ModSprite()
+            {
+                Margin = new MarginPadding(10);
+                Width = 60;
+                RelativeSizeAxes = Axes.Y;
+            }
+
+            [BackgroundDependencyLoader]
+            private void load(TextureStore textures)
+            {
+                var texture = textures.Get($"mods/{Mod}");
+
+                if (texture != null)
+                {
+                    Child = new Sprite
+                    {
+                        FillMode = FillMode.Fit,
+                        RelativeSizeAxes = Axes.Both,
+                        Anchor = Anchor.CentreRight,
+                        Origin = Anchor.CentreRight,
+                        Texture = texture
+                    };
+                }
+                else
+                {
+                    Mod selectedMod = null;
+
+                    switch (Mod)
+                    {
+                        case "DT":
+                            selectedMod = new OsuModDoubleTime();
+                            break;
+
+                        case "FL":
+                            selectedMod = new OsuModFlashlight();
+                            break;
+
+                        case "HT":
+                            selectedMod = new OsuModHalfTime();
+                            break;
+
+                        case "HD":
+                            selectedMod = new OsuModHidden();
+                            break;
+
+                        case "HR":
+                            selectedMod = new OsuModHardRock();
+                            break;
+
+                        case "NF":
+                            selectedMod = new OsuModNoFail();
+                            break;
+
+                        case "EZ":
+                            selectedMod = new OsuModEasy();
+                            break;
```

Comment:
should use `Mod.Acronym`

Reply:
resolving ruleset bindable and retrieve mod that matches acronym would be even more safe (for ruleset-specific mods to work as well).